addons/website_sale/static/tests/tours/website_sale_product_reviews_reactions_public.js

```diff
@@ -0,0 +1,30 @@
+import { registry } from "@web/core/registry";
+
+registry
+    .category('web_tour.tours')
+    .add('website_sale_product_reviews_reactions_public', {
+        url: '/shop?search=Storage Box Test',
+        steps: () => [
+            {
+                trigger: '.oe_product_cart a:contains("Storage Box Test")',
+                run: "click",
+            },
+            {
+                trigger: '.o_product_page_reviews_title',
+                run: "click",
+            },
+            {
+                trigger: "#chatterRoot:shadow .o-mail-Message-textContent:contains(Bad box!)",
+                run: "hover && click",
+            },
+            {
+                trigger: "#chatterRoot:shadow .o-mail-Message .o-mail-Message-actions",
+                run: async () => {
+                    const addReactionButton = document.querySelector('#chatterRoot').shadowRoot.querySelector("[title='Add a Reaction']")
+                    if (addReactionButton) {
+                        throw new Error("Public user is able to react");
+                    }
+                },
```

Comment:
Error messages that tells a wrong statement is not a well worded error message.
For clarity, it should be written as `throw new Error("Public users should not be able to add a reaction to a message")`.

"Public user" can also be ambiguous from technical code, because from the look of code (not an expert of port myself), technically the user is not authenticated but has token/hash/pid to let it know this is on behalf of a portal partner. What about saying "Non-authenticated user" instead? This part is not a big deal but just pointing out this could be misleading and using "authenticated" instead can be a bit more clear.